### ISSUE TITLE
Storyboards....scrolling too!

### ISSUE THREAD

**Issue by mbcoder17** [twotoasters/TWTSideMenuViewController#6]
Could you post an example with storyboards?

Also, in the luvocracy app, the menu appears to be a tableview, is this correct? If not, how would I get the buttons to scroll??

**Comment by jnjosh** [comment]
In the Luvocracy implementation it isn't a tableview, however the view can be anything. Since this menu view accepts two view controllers, all you need to do is give it a table view controller (or a view controller with a tableview in it) and you get that benefit.

I'll look into doing a sample with storyboards. I've never used them so it will be an interesting experiment.

**Comment by mbcoder17** [comment]
What is being used in the Luvocracy implementation then? I'm kind of new to this, but I really want to use this menu in my app, and I love the way it's used in Luvocracy...is it just a bunch of buttons in a scroll view?

Thanks.

**Comment by jnjosh** [comment]
Basically, yes. Buttons in a scroll view. I'm not sure that this is really the best route though. It made sense for us, but your results may vary. :)

**Comment by mbcoder17** [comment]
Ok...is there anyway you could go more in detail? Maybe in an email or private message? I'm very curious to know how you did it, I've been playing around with it but just can't get it. I'm also wondering how you made it jump back down under the status bar if you scroll too far above it...very cool!

**Comment by mbcoder17** [comment]
Hey, any update on the storyboard version?

Thanks!

**Comment by flightblog** [comment]
Do you absolutely need to use storyboards?

**Comment by mbcoder17** [comment]
Yes, my whole app is already using storyboards. I just need to change my menu so it matches iOS 7's design...

**Comment by leerie** [comment]
Hello, I'm going for a storyboard version too, but I think I can figure that part out. My question is on the Menu View Controller. I noticed in the sample code that in MenuViewController.viewDidLoad()  you have...

NSDictionary *viewDictionary = @{ @"imageView" : self.backgroundImageView };
    [self.view addConstraints:[NSLayoutConstraint constraintsWithVisualFormat:@"H:|-0-[imageView]" options:0 metrics:nil views:viewDictionary]];
    [self.view addConstraints:[NSLayoutConstraint constraintsWithVisualFormat:@"V:|-0-[imageView]" options:0 metrics:nil views:viewDictionary]];

My question is, since I'm planing on creating all of the views in Storyboard (or NIB for example), do I need to implement this in the view controller or forget about it. Basically, what does this do, I'm not this advanced yet. I figure its something to do with the autoLayout, but if I use a storyboard it looks like I'll be building most of its interface in code, as far as the background image.

**Comment by flightblog** [comment]
That code is handling the layout of the background image using auto layout. 

The @"H:|-0-[imageView]" constraint handles the horizontal layout and is pushing the background image to the lefthand edge and the @"V:|-0-[imageView]" constraint is push the background image to the very top.  

If you're handling the background color/image with your storyboard then you can leave this out.  (I haven't used storyboards).

**Comment by leerie** [comment]
Thanks for your reply. I'm gonna try it out without that code and see what happens.

**Comment by mbcoder17** [comment]
I got tired of waiting for a storyboards version, so I went menu searching....Here is a similar menu system that is compatible with storyboards right out of the box: https://github.com/romaonthego/RESideMenu

**Comment by leerie** [comment]
I've been looking at that one too, but honestly it seems so bland. The animations here are pretty nice but the read me needs some more information. Liking using multiple content views. Hopefully these guys will get into the storyboards and make life easier, but I think this one can work with it too. 

RESideMenu demo is kinda weird, like why not just use a UITableViewController instead of a regular view and manually setup a table. Prototype cells are more visual and easier to maintain than plist files, but those are minor gripes of mine.

The lack of Storyboard examples does scare folks away though.  Also, I will be adding some stuff in like ...

```
UISwipeGestureRecognizer *swipeLeft = [[UISwipeGestureRecognizer alloc] initWithTarget:self action:@selector(handleSwipeLeft:)];
[swipeLeft setDirection:UISwipeGestureRecognizerDirectionLeft];
[self.view addGestureRecognizer:swipeLeft];

UISwipeGestureRecognizer *swipeRight = [[UISwipeGestureRecognizer alloc] initWithTarget:self action:@selector(handleSwipeRight:)];
[swipeRight setDirection:UISwipeGestureRecognizerDirectionRight];
[self.view addGestureRecognizer:swipeRight];
```

Just not sure exactly where though, guessing to the mainContentController.

Honestly, these guys quickly replied, so I'm gonna give them a try and let you all know how it's working out.

**Comment by mbcoder17** [comment]
I’m using it in my app now, and I love it! Setup was extremely easy, and it doesn’t look half bad it my app either!

The reason I went searching for a different menu was the lack of a storyboard example here, but hopefully they can implement it soon!

**Comment by leerie** [comment]
Hmmm...Rocks and Hard places for me now :0 Decisions, Decisions.

**Comment by jnjosh** [comment]
Ok, guys. While I don't really understand why this needs a storyboard example to be useful, I will be working on some updates for this library today. I'll see if it is something I can throw a sample together. I don't use storyboards so I'll need to play with it some. Stay tuned.

**Comment by leerie** [comment]
Cause some people are not as smart as you, but we gotta get the same work done and you can't do it all bro. lol, I didn't think it was necessary, but the documentation is kinda slim. No worries.

**Comment by billinghamj** [comment]
Storyboards are used an awful lot & they make things extremely convenient. It isn't super trivial to make a library like this work with storyboards because the example works via the App Delegate, which is not how storyboard apps are designed to work

**Comment by scurioni** [comment]
Still no example with storyboards? That would be awesome!

**Comment by samuil-yanovski** [comment]
I'm a little late to the party, but in case anyone is still looking for a Storyboard example - I've put one [here](https://github.com/samuil-yanovski/twtsidemenu-storyboard).
